haystack/document_stores/weaviate.py

```diff
@@ -489,6 +489,9 @@ def write_documents(
                     # we "unnest" all value within "meta"
                     if "meta" in _doc.keys():
                         for k, v in _doc["meta"].items():
+                            assert (
+                                k not in _doc.keys()
+                            ), f'"meta" info contains duplicate key "{k}" with the top-level document structure'
```

Comment:
We prefer to raise explicit errors when reasonable. 

```suggestion
                            if k in _doc.keys():
                                raise ValueError(f'"meta" info contains duplicate key "{k}" with the top-level document structure')
```